nexus/posts/api.py

```diff
@@ -0,0 +1,65 @@
+from django.utils import timezone
+
+# Third Party Stuff
+from rest_framework import viewsets
+from rest_framework import mixins
+from rest_framework.decorators import action, parser_classes
+from rest_framework.parsers import FormParser, MultiPartParser
+from rest_framework.permissions import IsAdminUser, IsAuthenticated
+
+from nexus.base import response
+from . import models, serializers, permissions
+
+
+class PostViewSet(mixins.ListModelMixin, mixins.CreateModelMixin,
+                  mixins.UpdateModelMixin, mixins.RetrieveModelMixin,
+                  mixins.DestroyModelMixin, viewsets.GenericViewSet):
+    queryset = models.Post.objects.all()
+    permission_classes = (IsAuthenticated, permissions.IsAdminOrAuthorOfPost)
+
+    def get_serializer_class(self):
+        if self.action in ['approve', 'posted']:
+            return serializers.AdminPostSerializer
+        return serializers.PostSerializer
+
+    @action(methods=['PATCH'], detail=True, permission_classes=[IsAdminUser])
+    def approve(self, request, pk=None):
+        instance = self.get_object()
+        update = {'is_approved': True, 'approved_time': timezone.now()}
```

Comment:
let's rename this `data` or `payload` everywhere.

Reply:
ACK